rclcpp_lifecycle/CMakeLists.txt

```diff
@@ -28,12 +31,14 @@ target_include_directories(${PROJECT_NAME}
   PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
-# specific order: dependents before dependencies
-ament_target_dependencies(rclcpp_lifecycle
-  "rclcpp"
-  "rcl_lifecycle"
-  "lifecycle_msgs"
-  "rosidl_typesupport_cpp"
+target_link_libraries(${PROJECT_NAME}
+  ${lifecycle_msgs_TARGETS}
```

Comment:
unrelated, but is there any progress on being able to do use modern CMake targets like `lifecycle_msgs::lifecycle_msgs` ?